pkg/asset/agent/agentconfig/agenthosts.go

```diff
@@ -0,0 +1,307 @@
+
+func (a *AgentHosts) validateRoles(hostPath *field.Path, host agent.Host) field.ErrorList {
+	var allErrs field.ErrorList
+
+	if len(host.Role) > 0 && host.Role != masterRole && host.Role != workerRole {
+		allErrs = append(allErrs, field.Forbidden(hostPath.Child("Host"), "host role has incorrect value. Role must either be 'master' or 'worker'"))
+	}
+
+	return allErrs
+}
+
+func (a *AgentHosts) validateRendezvousIPNotWorker(rendezvousIP string, hosts []agent.Host) field.ErrorList {
+	var allErrs field.ErrorList
+
+	if rendezvousIP != "" {
+		for i, host := range hosts {
+			hostPath := field.NewPath("Hosts").Index(i)
+			if strings.Contains(string(host.NetworkConfig.Raw), rendezvousIP) && host.Role == workerRole {
+				errMsg := "Host " + host.Hostname + " has role 'worker' and has the rendezvousIP assigned to it. The rendezvousIP must be assigned to a control plane host."
+				allErrs = append(allErrs, field.Forbidden(hostPath.Child("Host"), errMsg))
+			}
+		}
+	}
+	return allErrs
+}
+
+// Add the baremetal hosts defined in install-config to the agent Hosts.
+func (a *AgentHosts) getInstallConfigDefaults(hosts []*baremetal.Host) error {
+	if hosts == nil {
+		return nil
+	}
+
+	for _, icHost := range hosts {
+		if icHost.BootMACAddress == "" {
+			return errors.New("host bootMACAddress is required")
+		}
+
+		host := agent.Host{
+			Hostname: icHost.Name,
+			Role:     icHost.Role,
+		}
+		if icHost.RootDeviceHints != nil {
+			host.RootDeviceHints = *icHost.RootDeviceHints
+		}
+		if icHost.NetworkConfig != nil {
+			contents, err := yaml.JSONToYAML(icHost.NetworkConfig.Raw)
+			if err != nil {
+				return errors.Wrap(err, "failed to unmarshal networkConfig")
+			}
+			host.NetworkConfig.Raw = contents
+			logrus.Warningf("hostNetworkconfig.Raw %s", icHost.NetworkConfig.Raw)
+			logrus.Warningf("contents %s", contents)
+
+			// Create interfaces table from NetworkConfig
+			var netInterfaces nmStateInterface
+			err = yaml.Unmarshal(contents, &netInterfaces)
+			if err != nil {
+				return fmt.Errorf("error unmarshalling NMStateConfig: %w", err)
+			}
+
+			var foundBootMac = false
+			for _, intf := range netInterfaces.Interfaces {
+				if intf.Name != "" && intf.MACAddress != "" {
+					hostInterface := &aiv1beta1.Interface{
+						Name:       intf.Name,
+						MacAddress: intf.MACAddress,
+					}
+					host.Interfaces = append(host.Interfaces, hostInterface)
+					if icHost.BootMACAddress == intf.MACAddress {
+						foundBootMac = true
+					}
+				}
+			}
+
+			if !foundBootMac {
+				logrus.Warnf("For host %s, BootMACAddress %s is not in NetworkConfig", icHost.Name, icHost.BootMACAddress)
+			}
+		}
+		if len(host.Interfaces) == 0 {
+			// Create interfaces table from BootMacAddress
+			hostInterface := &aiv1beta1.Interface{
+				Name:       "boot",
+				MacAddress: icHost.BootMACAddress,
+			}
+			host.Interfaces = append(host.Interfaces, hostInterface)
+		}
+
+		logrus.Infof("Using hosts from %s", agentAsset.InstallConfigFilename)
```

Comment:
This should be debug level IMHO. Also, this is inside the loop, so it should probably print the name of the host that it's using.

Reply:
Made this change in https://github.com/openshift/installer/pull/7645/commits/bade0514d466c7d1b12414970d0f6a99d2ab5ed6